### PR TITLE
Milestone 3. OPP-School-Library Project set-up-associations branch 

### PR DESCRIPTION
**Pull request by otmanTR**
In this branch following changes done:
-Created and updated Classroom class
-Created the has-many/belongs-to relationship between Classroom and Student
-Created and updated  Book class 
-Created and updated Rental class
-Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental
-Fixed linters

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals.push(rental)
+  end
```
Comment (ShadyShawkat) [inline book.rb]: - [x] Kindly create the new rental inside this method so you can use `self` to create a new rental for this book.

```ruby
def add_rental(person, date)
   Rental.new(date, self, person)
end
```

File: person.rb [hunk 2]
```diff
@@ -32,6 +34,10 @@ def of_age?
 
     false
   end
+
+  def add_book
+    @rentals << rental
+  end
```
Comment (ShadyShawkat) [inline person.rb]: - [x] Like the previous method of adding rentals, please create the new rental inside this method so you can use `self` to create a new rental for this person.

```ruby
def add_rental(book, date)
   Rental.new(date, book, self)
end
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @otmanTR, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (otmanTR) [reply person.rb]: Thank you.

File: book.rb (hunk 1 above)
Comment (otmanTR) [reply book.rb]: Thank you.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
+    Rental.new(date, person, self)
+  end
```
Comment (Gambit142) [inline book.rb]: - [x] Great work implementing the changes suggested by the previous reviewer. There is a little change you need to make. Due to the parameters passed to the `initialize` method of the `Rental` class, there would be an error when creating a new instance of the `Rental` class in the `add_rental` method of the `Book` class. In the `initialize` method of the `Rental` class, you passed `date`, `book` and `person`: ` def initialize(date, book, person)` therefore when creating the instance in the Book class, it should be `Rental.new(date, self, person)`. Kindly correct this. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @otmanTR 👋, 

Great work on making the changes requested by a previous reviewer 👏 

## Highlights
- All linters checks passed ✔️ 
- README is professional ✔️ 
- Established associations between classes ✔️ 

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 3 above)
Comment (otmanTR) [reply book.rb]: Thank you, I missed this key.

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @otmanTR , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Gitflow is used correctly :white_check_mark: 
- [x] Associations setup correctly :white_check_mark: 
- [x] Pull request has a descriptive summary and title :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**